meetings/working-groups/field-keyword/FK-2024-08-07 Nullability analysis with the `field` keyword.md

```diff
@@ -0,0 +1,373 @@
+# Nullability analysis with the `field` keyword
+
+Table of contents:
+
+- [Motivation](#motivation)
+- [Introduction](#introduction)
+- [Constructor initialization](#constructor-initialization)
+- [Field-targeted nullability attributes](#field-targeted-nullability-attributes)
+- [Property-targeted nullability attributes](#property-targeted-nullability-attributes)
+- [Null-resilient getters](#null-resilient-getters)
+- [Alternatives](#alternatives)
+
+## Motivation
+
+Some vastly common patterns which are served by the `field` feature are lazy initialization, and defaulting. When a field-backed property is non-nullable, it's required to be initialized during construction. This flies in the face of these kinds of properties. They are well-formed. Warnings requiring constructor initialization would be noisy and unwelcome in these properties and in many similar forms.
+
+```cs
+public List<int> LazyProp => field ??= new();
+
+public string DefaultingProp { get => field ?? GetDefault(); private set; }
+```
+
+These scenarios are made possible using the existing nullability attributes, like `[MaybeNull]`, just as they would be for manually-declared fields.
+
+We'll walk first through enabling the user to provide these nullability attributes explicitly for the backing field, and then through how the compiler builds on this capability to apply these attributes automatically in a simple and intuitive fashion.
```

Comment:
i would like to add a comment that these use cases are not speculative.  They were a large part of the examples and conversation around `field` (and feel free to link to the discussion page there).  

Basically, i want to sidestep people going: this isn't a big deal, we don't need to do anything here.

rather, it's a very big deal.  People have been expecting these scenarios to work as part of the entire 5+ year journey this feature has been on.  Not having it would be a bitter pill.